cassandraprotocol/frame/decoder.go

```diff
@@ -117,6 +130,21 @@ func (c *Codec) DiscardBody(bodyLength int32, source io.Reader) (err error) {
 	return err
 }
 
+// ReadBody reads the contents of a frame body without decoding them. It is illegal to call this method before calling
+// DecodeHeader.
+func (c *Codec) ReadBody(bodyLength int32, source io.Reader) (body []byte, err error) {
+	if bodyLength < 0 {
+		return nil, fmt.Errorf("invalid body length: %d", bodyLength)
+	} else if bodyLength == 0 {
+		return []byte{}, nil
+	}
+
+	count := int64(bodyLength)
+	buf := &bytes.Buffer{}
+	_, err = io.CopyN(buf, source, count)
+	return buf.Bytes(), err
```

Comment:
```suggestion
	if n, err = io.CopyN(buf, source, count); err != nil {
		return nil, fmt.Errorf("cannot read body: %w", err)
	} else if n != count {
		return nil, fmt.Errorf("body length mismatch, expected %d, got: %d", count, n)
	}
	return buf.Bytes(), nil
```

Reply:
the docs specify that `n != count` can only happen when `err != nil` so this doesn't seem necessary

Reply:
Yeah and if that is the case then the "else if" clause would not be reached as it would return before getting there. 

If the intention here is to give a clearer, more specific error message in case of length mismatch we could enrich the error message returned in the "err != nil" clause.

Reply:
You mean something like this, right? Yeah I'm ok with it
```
if n, err = io.CopyN(buf, source, count); err != nil {
		return nil, fmt.Errorf("cannot read body: %w, bytes read: %d", err, n)
	}
	return buf.Bytes(), nil
```

Reply:
done